user/TASK/gimbal_task/gimbal_task.c

```diff
@@ -121,24 +172,63 @@ static void get_new_data(Gimbal_t *gimbal_data){
  */
 static void update_setpoints(Gimbal_t *gimbal_set){
     
-    gimbal_set->yaw_motor.pos_set += (-1) * int16_deadzone(gimbal_set->rc_update->rc.ch[2], -DEADBAND, DEADBAND) / 10;
-    gimbal_set->pitch_motor.pos_set += int16_deadzone(gimbal_set->rc_update->rc.ch[3], -DEADBAND, DEADBAND) / 10;
+    //yaw:
+    //rc -> theta -> complex rotation -> new setpoint
+
+    // TODO: FIX
+    if(gimbal_set->rc_update->rc.s[0] == RC_SW_MID || gimbal_set->rc_update->rc.s[0] == RC_SW_UP){
+        fp32 theta = -1 * int16_deadzone(gimbal_set->rc_update->rc.ch[2], -DEADBAND, DEADBAND)
+                * Motor_Ecd_to_Rad / 80.0f;
+        fp32 rotation[2] = {cos(theta), sin(theta)}; // TODO: consider alternative {cos{theta}, sin{theta}}
+        multiply_complex_a_by_b(gimbal_set->yaw_setpoint, rotation);
+        make_unit_length(gimbal_set->yaw_setpoint);
+
+        gimbal_set->pitch_motor.pos_set += int16_deadzone(gimbal_set->rc_update->rc.ch[3], -DEADBAND, DEADBAND) / 80.0f;
+    }
     
-    int16_constrain(gimbal_set->yaw_motor.pos_set, YAW_MIN, YAW_MAX);
-    int16_constrain(gimbal_set->pitch_motor.pos_set, PITCH_MIN, PITCH_MAX);
+    char print[30];
+    sprintf(print, "Constrain pitch %d between %d, %d \n\r", gimbal_set->pitch_motor.pos_set, PITCH_MIN, PITCH_MAX);
+    serial_send_string(print);
+    //int16_constrain(gimbal_set->yaw_motor.pos_set, YAW_MIN, YAW_MAX);
+    gimbal_set->pitch_motor.pos_set = int16_constrain(gimbal_set->pitch_motor.pos_set, PITCH_MIN, PITCH_MAX);
+    sprintf(print, "Constrained to %d \n\r", gimbal_set->pitch_motor.pos_set);
+    serial_send_string(print);
```

Comment:
Should be inside of a if (DEBUG_SETPOINTS) block

Reply:
Deleted